contracts/Governance/ACMCommandsAggregator.sol

```diff
@@ -0,0 +1,113 @@
+// SPDX-License-Identifier: BSD-3-Clause
+pragma solidity 0.8.25;
+
+import { IAccessControlManagerV8 } from "../Governance/IAccessControlManagerV8.sol";
+
+contract ACMCommandsAggregator {
+    /*
+     * @notice Enum to differentiate between giving and revoking permissions
+     */
+    enum PermissionType {
+        GIVE,
+        REVOKE
+    }
+
+    /*
+     * @notice Struct to store permission details
+     */
+    struct Permission {
+        /*
+         * @notice Type of permission
+         */
+        PermissionType permissionType;
+        /*
+         * @notice Address of the contract
+         */
+        address contractAddress;
+        /*
+         * @notice Function signature
+         */
+        string functionSig;
+        /*
+         * @notice Address of the account
+         */
+        address account;
+    }
+
+    /**
+     * @notice Access control manager contract
+     */
+    IAccessControlManagerV8 public immutable ACM;
+
+    /*
+     * @notice Mapping to store permissions
+     */
+    mapping(uint256 => Permission[]) public permissions;
+
+    /*
+     * @notice Index for the next permissions
+     */
+    uint256 public nextIndex;
```

Comment:
`permissions` could be an array, instead of a mapping. In that case, we wouldn't need `nextIndex` because we could read `permissions.length`. Do we have any constraint to not use an array here? I don't have a strong opinion about this, btw